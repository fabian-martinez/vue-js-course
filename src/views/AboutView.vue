<template>
  <div class="about">
    <h1>{{ title }}</h1>
    <button @click="changeTitle">Cambiar titulo</button>
  </div>
</template>

<script>
import { 
  onMounted, onUnmounted, onBeforeMount, 
  onBeforeUpdate, onBeforeUnmount, onErrorCaptured, 
  onRenderTracked, onRenderTriggered, onActivated, 
  onDeactivated, onServerPrefetch, onUpdated, ref,  } from "vue";
export default {
  setup() {

    const title = ref('About Page')

    console.log('Setup se llama');
    onMounted(() => { console.log('Mounted se llamó');})
    onUnmounted(() => { console.log('Unmounted se llamó');})
    onBeforeMount(() => { console.log('BeforeMount se llamó');})
    onBeforeUpdate(() => { console.log('BeforeUpdate se llamó');})
    onBeforeUnmount(() => { console.log('BeforeUnmount se llamó');})
    onErrorCaptured(() => { console.log('ErrorCaptured se llamó');})
    onRenderTracked(() => { console.log('RenderTracked se llamó');})
    onRenderTriggered(() => { console.log('RenderTriggered se llamó');})
    onActivated(() => { console.log('Activated se llamó');})
    onDeactivated(() => { console.log('Deactivated se llamó');})
    onServerPrefetch(() => { console.log('ServerPrefetch se llamó');})
    onUpdated(() => { console.log('Updated se llamó');})

    return{
      title,
      changeTitle: () => {title.value='About Page ha cambiado'}
    }
  }

}
</script>
