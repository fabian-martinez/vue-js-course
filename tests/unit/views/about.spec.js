import { shallowMount } from '@vue/test-utils';
import AboutView from '@/views/AboutView';

describe('Pruebas en el about view ', () => {
    it('debe de hacer match con el snapshop cuando cargen el about', () => {
        const wrapper = shallowMount(AboutView)

        expect(wrapper.html()).toMatchSnapshot()
    });
});