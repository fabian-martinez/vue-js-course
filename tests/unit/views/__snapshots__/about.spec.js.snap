// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page debe de hacer match con el snapshop cuando cargen el about 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Pruebas en el about view  debe de hacer match con el snapshop cuando cargen el about 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Pruebas en el about view debe de hacer match con el snapshop cuando cargen el about 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;
